client/app/scripts/utils/web-api-utils.js

```diff
@@ -58,8 +58,14 @@ export function basePathSlash(urlPath) {
   return `${basePath(urlPath)}/`;
 }
 
+const API_PATH = basePathSlash(window.location.pathname) === '/'
+  ? 'api'
+  : `/api${window.location.pathname}/api`;
+
 const wsProto = location.protocol === 'https:' ? 'wss' : 'ws';
-const wsUrl = `${wsProto}://${location.host}${basePath(location.pathname)}`;
+export const wsUrl = basePathSlash(window.location.pathname) === '/'
+  ? `${wsProto}://${location.host}${basePath(window.location.pathname)}`
+  : `${wsProto}://${location.host}/api${basePath(window.location.pathname)}`;
```

Comment:
I was thinking the `web-api-utils` could use some re-factoring; maybe replace the `reqwest` module with a decorated XHR function that already has the hostname and path stored. We may be able to use the `redux-thunk` .withExtraArgument() to pass down deps to async actions.